### DISCUSSION TITLE
Set width of excel columns

### DISCUSSION THREAD

**Discussion by ryanyanzhi** [ng-alain/ng-alain#2040] [Show and tell]
If you need to set column width using xlsx service, you can add this as following to export method parameters:
```typescript
callback: (wb) => {
     wb.Sheets.sheet['!cols'] = [];
     for (const col of this.columns) {
          wb.Sheets.sheet['!cols'].push({ wch: col.width });
     }
 }
```
the `width` is set here:
```typescript
columns: STColumn[] = [
    { title: 'price', index: 'price', width: 12 },
]
```

References:
https://ng-alain.com/components/xlsx/zh
https://segmentfault.com/a/1190000021272653
https://github.com/SheetJS/sheetjs

**Comment by cipchk** [comment]
@Martin367 Good, can you provide an issues, i thank this is a bug.
